tests/unit/apps/test_auxiliary.py

```diff
@@ -1074,3 +1134,107 @@ def test_auxiliary_upgrade_by_unit(mock_super, model):
     # Parent class was called with units=None even that units were passed to the
     # Auxiliary app, meaning that will create all-in-one upgrade strategy
     mock_super.assert_called_with(target, False, None)
+
+
+@pytest.mark.asyncio
+@patch("cou.utils.openstack.OpenStackCodenameLookup.find_compatible_versions")
+async def test_ceph_osd_verify_nova_compute_pass(mock_lookup, model):
+    """Test Ceph-osd verifying all nova computes."""
+    target = OpenStackRelease("victoria")
+    mock_lookup.return_value = [target]
+    nova_compute = MagicMock(spec_set=NovaCompute)()
+    nova_compute.charm = "nova-compute"
+    nova_compute.units = {"nova-compute/0": COUUnit("nova-compute/0", None, "22.0.0")}
+    app = CephOsd(
+        name="ceph-osd",
+        can_upgrade_to="octopus/stable",
+        charm="ceph-osd",
+        channel="octopus/stable",
+        config={"source": {"value": "distro"}},
+        machines={},
+        model=model,
+        origin="ch",
+        series="focal",
+        subordinate_to=[],
+        units={},
+        workload_version="17.0.1",
+    )
+    model.get_applications.return_value = [app, nova_compute]
+
+    await app._verify_nova_compute(target)
+
+    model.get_applications.assert_awaited_once_with()
+    mock_lookup.assert_called_once_with("nova-compute", "22.0.0")
+
+
+@pytest.mark.asyncio
+@patch("cou.utils.openstack.OpenStackCodenameLookup.find_compatible_versions")
+async def test_ceph_osd_verify_nova_compute_fail(mock_lookup, model):
+    """Test Ceph-osd verifying all nova computes."""
+    mock_lookup.return_value = [OpenStackRelease("ussuri")]
+    target = OpenStackRelease("victoria")
+    nova_compute = MagicMock(spec_set=NovaCompute)()
+    nova_compute.charm = "nova-compute"
+    nova_compute.units = {"nova-compute/0": COUUnit("nova-compute/0", None, "22.0.0")}
```

Comment:
nit: I know that mocking `find_compatible_versions` the workload_version is irrelevant, but changing to the ussuri workload_version is preferable.
```suggestion
    nova_compute.units = {"nova-compute/0": COUUnit("nova-compute/0", None, "21.0.0")}
```

Reply:
Yes, I accidentally change it base on this [comment](https://github.com/canonical/charmed-openstack-upgrader/pull/280#discussion_r1518108139), if there will be another change-request I will change it. Thanks